src/jllvm/vm/VirtualMachine.cpp

```diff
@@ -200,20 +200,12 @@ jllvm::VirtualMachine::VirtualMachine(BootOptions&& bootOptions)
         std::pair{"jllvm_build_class_cast_exception",
                   [&](Object* object, ClassObject* classObject) -> Object*
                   {
-                      llvm::StringRef className = object->getClass()->getClassName();
-                      llvm::StringRef prefix;
-                      std::string name;
-                      if (classObject->isClass() || classObject->isInterface())
-                      {
-                          prefix = "class";
-                          name = classObject->getClassName();
-                      }
-                      else
-                      {
-                          // TODO: Pretty print array
-                      }
+                      std::string className = FieldType{object->getClass()->getDescriptor()}.pretty();
+                      std::string name = FieldType{classObject->getDescriptor()}.pretty();
```

Comment:
Doesn't `getDescriptor` return a `FieldType` that you can just call `pretty` on? This looks like a noop constructor call to me.